### ISSUE TITLE
Add ":set" Summary Page

### ISSUE THREAD

**Issue by flatcap** [neomutt/neomutt#1178]
This is an "easy" task ([find more](https://github.com/neomutt/neomutt/issues?q=label:"good%20first%20issue"+is:open)) and it's open to everyone.  [_read more..._](https://neomutt.org/dev/easy-tasks)

## Description

We'd like to add a set of summary pages to NeoMutt.  See #162 for our ideas.

Much of the work has already been done in the [sample/summary](https://github.com/neomutt/neomutt/tree/sample/summary) branch.
This code has the infrastructure to support `:` commands.

### Part 1: `:set`

Show all the config variables on a page.

Create a page and format all of the config variables onto it.

### Part 2: `:set all`

Change the `set` command to only display the config that the user has changed.
This means comparing all the values against the initial values.

Only show **all** the config if the user runs `:set all`

**Comment by yuzibo** [comment]
Hello,
I typed *:set* in Neomutt and i got nothing.What am i missing?
Thanks,

**Comment by flatcap** [comment]
It's isn't implemented yet, that's the task.
Make NeoMutt behave like Vim's `:set`

**Comment by yuzibo** [comment]
Hello,
Please could you tell me the issue bases which branch?
And i have seen @christopher-john-czettel that have done almost work.If base master branch, which is not good idea i think

**Comment by flatcap** [comment]
I've just rebased [sample/summary] to master.

The original branch contains test code that we'd want to replace,
so you could clone that to [devel/summary] and work there.

**Comment by yuzibo** [comment]
I am sorry that you mean  i can "git checkout -b devel/summary origin/master"?
In fact, i "git checkout -b devel/summary origin/sample/summary",
But, i have faced unexpected halt several  times after make && make install .
It looks like halting(disconnection with imap server), but can type command,but it report:
"can not copy message" in loop
And  fail to quit it.

**Comment by flatcap** [comment]
We don't want to checkout master, we want to start from the summary branch.

```sh
# make sure everything's up-to-date
git fetch origin

# create a devel copy of the summary branch
git checkout -b devel/summary origin/sample/summary

# its 'upstream' will still be the summary branch, until we push
git push -u origin devel/summary
```

> But, i have faced unexpected halt several times after make && make install .
> It looks like halting(disconnection with imap server), it can type command,but it report:

I have no idea what's going wrong here.
The summary branch doesn't change any of the mailbox code.

**Comment by flatcap** [comment]
Implemented by 908a11fa1
